js/join.js

```diff
@@ -0,0 +1,335 @@
+  dataValidator(skills, 6);
+  dataValidator(college, 5);
+  dataValidator(forFun, 100);
+  dataValidator(funFact, 100);
+  dataValidator(whyRds, 100);
+}
+
+function previewFiller() {
+  previewFName.innerText = window.localStorage.getItem('firstName');
+  previewLName.innerHTML = window.localStorage.getItem('lastName');
+  previewCity.innerHTML = window.localStorage.getItem('city');
+  previewState.innerHTML = window.localStorage.getItem('state');
+  previewCountry.innerHTML = window.localStorage.getItem('country');
+  previewIntro.innerHTML = window.localStorage.getItem('introduction');
+  previewSkills.innerHTML = window.localStorage.getItem('skills');
+  previewInstitution.innerHTML = window.localStorage.getItem('college');
+  previewForFun.innerHTML = window.localStorage.getItem('forFun');
+  previewFunFact.innerHTML = window.localStorage.getItem('funFact');
+  previewWhyRds.innerHTML = window.localStorage.getItem('whyRds');
+  previewHeardAbout.innerHTML = window.localStorage.getItem('heardAbout');
+}
+
+//Direct to the page user left from
+window.addEventListener('load', () => {
+  const currentFlowState = window.localStorage.getItem('flowState');
+  showPage(currentFlowState);
+  autoFillTheFields();
+  toggleNextButton();
+  if (currentFlowState == flowState.previewPage) {
+    previewFiller();
+  }
+});
+
+//initializer
+if (!window.localStorage.getItem('flowState')) {
+  window.localStorage.setItem('flowState', flowState.notStarted);
+}
+
+//Value updaters
+city.addEventListener('input', () => {
+  window.localStorage.setItem('city', city.value);
+  toggleNextButton();
+});
+
+state.addEventListener('input', () => {
+  window.localStorage.setItem('state', state.value);
+  toggleNextButton();
+});
+
+country.addEventListener('input', () => {
+  window.localStorage.setItem('country', country.value);
+  toggleNextButton();
+});
+
+introduction.addEventListener('input', () => {
+  window.localStorage.setItem('introduction', introduction.value);
+  toggleNextButton();
+  dataValidator(introduction, 100);
+});
+
+skills.addEventListener('input', () => {
+  window.localStorage.setItem('skills', skills.value);
+  toggleNextButton();
+  dataValidator(skills, 6);
+});
+
+college.addEventListener('input', () => {
+  window.localStorage.setItem('college', college.value);
+  toggleNextButton();
+  dataValidator(college, 5);
+});
+
+forFun.addEventListener('input', () => {
+  window.localStorage.setItem('forFun', forFun.value);
+  toggleNextButton();
+  dataValidator(forFun, 100);
+});
+
+funFact.addEventListener('input', () => {
+  window.localStorage.setItem('funFact', funFact.value);
+  toggleNextButton();
+  dataValidator(funFact, 100);
+});
+
+whyRds.addEventListener('input', () => {
+  window.localStorage.setItem('whyRds', whyRds.value);
+  toggleNextButton();
+  dataValidator(whyRds, 100);
+});
+
+heardAbout.addEventListener('input', () => {
+  window.localStorage.setItem('heardAbout', heardAbout.value);
+  toggleNextButton();
+});
```

Comment:
can we extract this into a function that takes an array of elements as an argument and adds the `input event listener` to each element? 
This is so we avoid repeating the same code multiple times.

Reply:
we are currently updating the local storage as soon as a change in the input is made, I think we can debounce the callback function of the event listener to avoid making very frequent changes to our local store.